### ISSUE TITLE
TODO: change name of widget in the system tray (instead of scriptinator)

### ISSUE THREAD

**Issue by tubbadu** [tubbadu/Scriptinator#9]

**Comment by Sadi58** [comment]
It would be great **if possible**!
At the moment, you can change the name in the file "metadata.desktop" only, but it changes the name of the widget as a whole, and not just one instance of it.

**Comment by tubbadu** [comment]
exactly, currently I have 4 copies of scriptinator because I need 4 different system trays, but It would be better if all this could be done with the GUI plasmoid setting, without touching the source code

I'll look forward to it!

**Comment by tubbadu** [comment]
Here's my attempt to achieve this (and #8 as well):
the "original" scriptinator will *not* be in the system tray. I will add a new tab in the plugin configuration window, something like "tray instances", with an UI similar to this:

![scriptinator](https://user-images.githubusercontent.com/56864069/215278419-dbf9d855-38f4-4a57-b0d0-1ee0d16dd4a6.png)

here the working mechanism: It will read all folders names in the location `$HOME/.local/share/plasma/plasmoid/`, with the filter `com.github.scriptinator.*`
each of these folder will contains just one single metadata.desktop file like this:
```
[Desktop Entry]
Name=NAME
Comment=DESCRIPTION
Icon=ICON
X-KDE-PluginInfo-Name=com.github.scriptinator.PLUGIN_NAME



Type=Service
X-KDE-ServiceTypes=Plasma/Applet
X-Plasma-API=declarativeappletscript
X-Plasma-MainScript=ui/main.qml
X-KDE-PluginInfo-Author=tubbadu
X-KDE-PluginInfo-Email=tubbadu@gmail.com
X-KDE-PluginInfo-Version=1.1
X-KDE-PluginInfo-Website=https://github.com/tubbadu/Scriptinator
X-KDE-PluginInfo-Category=Utilities
X-KDE-PluginInfo-License=GPL
X-Plasma-NotificationArea=true
X-Plasma-NotificationAreaCategory=SystemServices
X-Plasma-RootPath=com.github.scriptinator
```

when hitting the "save" button, it will update the 4 fields at the beginning, allowing us to change name, tooltip description, and icon of the tray instance. the plugin_name field perhaps is redundant, it's better to manage it automatically to avoid problems

doing this, any scriptinator "clone" will be found in the system tray (should be manually enabled, I don't know if it's possible to set it as automatically enabled) but also in the "add plasmoid" popup, which is not good, as it will just be a scriptinator clone with a different icon and name... Is this acceptable?

@Sadi58 let me know your thoughts about this!

**Comment by Sadi58** [comment]
All this sounds very good to me.
Unfortunately I have only some basic shell scripting skills, so I can't help with these matters any more than just throwing some ideas:
I think things could be simpler if you could achieve the use of multiple instances of Scriptinator in system tray by working in one folder: `$HOME/.local/share/plasma/plasmoids/com.github.scriptinator`, e.g. adding different `metadata.desktop` files for individual instances of Scriptinator under the sub-folder `contents` (or under different sub-folders with a certain name pattern like `plugin_<name>`), instead of creating different plasmoid folders named `com.github.scriptinator.*`.

**Comment by tubbadu** [comment]
That would surely be tidier, I'll do my researches to understand if we can do it
